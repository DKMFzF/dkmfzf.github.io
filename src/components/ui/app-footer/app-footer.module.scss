@use '~styles/variables' as *;

.footer {
	width: 100%;
	max-width: $max-page-width;
	margin: 0 auto;
	color: $bg-primary-color;
	font-family: 'Segoe UI', 'Tahoma', 'Geneva', 'Verdana', sans-serif;
	background-color: transparent;

	&__bottom {
		display: flex;
		justify-content: end;
		gap: 20px;
		width: 100%;
		text-align: end;
		border-top: 1px solid rgb(255 255 255 / 10%);
	}

	&__text {
		margin: 0;
		color: $bg-primary-color;
		font-size: 0.9rem;
	}
}

.content {
	display: flex;
	align-items: center;
	margin: 0 auto;
	padding: 40px;
	font-family: $font-title;
	text-transform: uppercase;
	opacity: $opacity;
	margin-inline: 25px;
}

.links {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
	gap: 15px;
	margin-bottom: 1.5rem;
}

.link {
	color: $bg-primary-color;
	font-size: 1rem;
	text-decoration: none;
	transition: color 0.3s ease;

	&:hover {
		color: #3498db;
	}
}

.social {
	display: flex;
	justify-content: center;
	gap: 15px;
}

.socialLink {
	color: $bg-primary-color;
	font-size: 1rem;
	text-decoration: none;
	opacity: $opacity;
	transition: opacity $time-transition;

	&:hover {
		opacity: 1;
	}
}

/* Адаптивность */

@media (width <= 768px) {
	.links {
		flex-direction: column;
		align-items: center;
		gap: 5px;
	}

	.social {
		flex-wrap: wrap;
		margin-bottom: 1rem;
	}
}
