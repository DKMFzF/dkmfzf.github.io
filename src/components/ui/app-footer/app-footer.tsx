import { FC } from 'react';
import styles from './app-footer.module.scss';

export const AppFooterUI: FC = () => {
	const currentYear = new Date().getFullYear();

	return (
		<footer className={styles.footer}>
			<div className={styles.content}>
				<div className={styles.social}>
					<a
						href='https://github.com/DKMFZF'
						target='_blank'
						rel='noopener noreferrer'
						aria-label='GitHub'
						className={styles.socialLink}
					>
						GitHub
					</a>
					<a
						href='https://vk.com/dkmfzf'
						target='_blank'
						rel='noopener noreferrer'
						aria-label='LinkedIn'
						className={styles.socialLink}
					>
						VK
					</a>
					<a
						href='https://habr.com/ru/users/DKMFzF/'
						target='_blank'
						rel='noopener noreferrer'
						aria-label='Twitter'
						className={styles.socialLink}
					>
						Habr
					</a>
					<a
						href='https://t.me/DKMFZF'
						aria-label='Email'
						className={styles.socialLink}
					>
						Tg
					</a>
				</div>
				<div className={styles.bottom}>
					<p>&copy; {currentYear} My portfolio</p>
				</div>
			</div>
		</footer>
	);
};
