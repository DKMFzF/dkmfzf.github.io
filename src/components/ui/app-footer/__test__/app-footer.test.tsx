import { describe, it, beforeEach, afterEach, jest } from '@jest/globals';
import { testRenderComponent, testCurrentText } from '@utils-testing';
import { AppFooterUI } from '../app-footer';

describe('[AppFooterUI]', () => {
	beforeEach(() => {
		jest.spyOn(Date.prototype, 'getFullYear').mockReturnValue(2025);
	});

	afterEach(() => {
		jest.restoreAllMocks();
	});

	it('Renders footer correctly and matches snapshot', () =>
		testRenderComponent(<AppFooterUI />));

	it('Displays current year correctly', () =>
		testCurrentText(<AppFooterUI />, 'years', 'Â© 2025 DKMFZF PORTFOLIO'));
});
