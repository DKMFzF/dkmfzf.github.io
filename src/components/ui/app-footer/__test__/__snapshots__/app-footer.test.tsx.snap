// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AppFooterUI] Renders footer correctly and matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer__content"
    >
      <div
        class="footer__social"
      >
        <a
          aria-label="GitHub"
          class="footer__social-link"
          href="https://github.com/DKMFZF"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          aria-label="LinkedIn"
          class="footer__social-link"
          href="https://vk.com/dkmfzf"
          rel="noopener noreferrer"
          target="_blank"
        >
          VK
        </a>
        <a
          aria-label="Twitter"
          class="footer__social-link"
          href="https://habr.com/ru/users/DKMFzF/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Habr
        </a>
        <a
          aria-label="Email"
          class="footer__social-link"
          href="https://t.me/DKMFZF"
        >
          Tg
        </a>
      </div>
      <div
        class="footer__bottom"
      >
        <span
          class="footer__text"
        >
          VERSION 1.40.1
        </span>
        <p
          data-testid="years"
        >
          Â© 2025 DKMFZF PORTFOLIO
        </p>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
